src/dials/command_line/dials_import.py

```diff
@@ -896,23 +898,21 @@ def do_import(
         num_images += len(e.imageset)
         counted_imagesets.append(e.imageset)
 
-    format_list = {str(e.imageset.get_format_class()) for e in experiments}
-    try:
-        template_list = {
-            str(e.imageset.get_template())
-            + ":{0}:{1}".format(*e.scan.get_image_range())
-            for e in experiments
-        }
-    except AttributeError:
-        # For stills we need a different approach
-        template_list = {}
-
+    unique_formats = OrderedSet()
+    unique_templates = OrderedSet()
+    for imgset in counted_imagesets:
+        unique_formats.add(imgset.get_format_class())
+        if scan := imgset.get_scan():
+            start, end = scan.get_image_range()
+            unique_templates.add(f"{imgset.get_template()}:{start}:{end}")
+        else:
+            unique_templates.add(f"{imgset.reader().paths()[0]}")
     # Print out some bulk info
     logger.info("-" * 80)
-    for f in format_list:
-        logger.info("  format: %s", f)
-    for f in template_list:
-        logger.info("  template: %s", f)
+    for fmt in unique_formats:
+        logger.info("  format: %s", fmt)
+    for template in unique_templates:
```

Comment:
Hobgoblin:

I understand why `fmt <- format` but it's annoying to have `template` spelled in full

Reply:
In fairness, I don't have a better suggestion than using `f` and `t` so 🤷‍♂️